File: OOP/tic-tac-toe-board-game-app/src/com/techlab/board/Board.java
package com.techlab.board;

import com.techlab.Exception.AllReadyMarkCellException;

public class Board {
	private Cell[] boardCells;
	private int MAXIMUM_POSITION_BOARD = 9;

	public Board() {
		boardCells = new Cell[9];
		for (int index = 0; index < MAXIMUM_POSITION_BOARD; index++) {
			boardCells[index] = new Cell();
		}
	}

	public void setBoardCellMark(int position, Mark mark) throws AllReadyMarkCellException {
		if (!checkBoardCellPosition(position))
			throw new IndexOutOfBoundsException();
		boardCells[position].setCellState(mark);
	}

	public Cell getBoardCellMark(int position) {
		return boardCells[position];
	}

	public Cell[] getBoardState() {
		return boardCells;
	}

	private boolean checkBoardCellPosition(int position) {
		if (position < MAXIMUM_POSITION_BOARD)
			return true;
		return false;
	}
}

File: nodeJs/tic-tac-toe-board-game/board.js
let Cell = require('./cell');

"use strict";
class Board {
    constructor() {
        this.MAXIMUM_INDEX_BOARD = 9;
        this._boardCells = [];
        for (let index = 0; index < this.MAXIMUM_INDEX_BOARD; index++) {
            this._boardCells.push(new Cell());
        }
    }
    setBoardCellMark(index, mark) {
        if (index >= this.MAXIMUM_INDEX_BOARD)
            throw new Error('index out of bound');
        this._boardCells[index].cellState = mark;
    }
    getBoardCellMark(index) {
        return this._boardCells[index];
    }
    getBoardState() {
        return this._boardCells;
    }
    displayBoard() {
        console.log("\n\t -----------board---------------");

        console.log("\t\t" + this._boardCells[0].cellState + "   | " + this._boardCells[1].cellState + "  | " + this._boardCells[2].cellState);
        console.log("\t\t ------------ ");
        console.log("\t\t" + this._boardCells[3].cellState + "   | " + this._boardCells[4].cellState + "  | " + this._boardCells[5].cellState);
        console.log("\t\t ------------ ");
        console.log("\t\t" + this._boardCells[6].cellState + "   | " + this._boardCells[7].cellState + "  | " + this._boardCells[8].cellState + "\n");
    }
}
module.exports = Board;